### PR TITLE
Project initial setup, packages created and dependencies added.

### PR DESCRIPTION
**Pull request by doersweb**
This commit does the following 
1. Added buildSrc and created required dependency objects.
2. build.gradle to build.gradle.kts
3. Basic packages created.
4. interactors created - for both authentication and events management

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    implementation(Compose.hiltNavigation)
+    implementation(Compose.material)
+    implementation(Compose.runtime)
+    implementation(Compose.navigation)
+    implementation(Compose.viewModel)
+    implementation(Compose.activity)
+
+    implementation(Core.coreKtx)
+    implementation(DaggerHilt.hiltAndroid)
+    kapt(DaggerHilt.hiltCompiler)
```
Comment (philipplackner) [inline app/build.gradle.kts]: I recommend to use KSP (Kotlin Symbol Processing), which is a faster alternative to kapt. Recently, Dagger and Room got supported for that.

I explain it here: https://www.youtube.com/watch?v=WVhgUGyUnhQ

File: app/src/main/java/com/doersweb/tasky/interactor/authentication/AuthenticationInteractorImpl.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    override suspend fun registerUser(fullName: String, email: String, password: String) {
+        TODO("Not yet implemented")
+    }
+
+    override suspend fun signIn(email: String, password: String) {
+        TODO("Not yet implemented")
+    }
+
+}
```
Comment (philipplackner) [inline app/src/main/java/com/doersweb/tasky/interactor/authentication/AuthenticationInteractorImpl.kt:16]: What's your definition of interceptor?

I'm assuming something like a bundled use case - in that case what's the benefit of abstracting it? An abstraction only makes sense when there are at least 2 (planned) implementations. For example, one for the production implementation and one for testing - but in my experience you never need to replace business logic for testing.

**Review by philipplackner — APPROVED** [review]

File: app/src/main/java/com/doersweb/tasky/interactor/authentication/AuthenticationInteractorImpl.kt (hunk 2 above)
Comment (doersweb) [reply app/src/main/java/com/doersweb/tasky/interactor/authentication/AuthenticationInteractorImpl.kt:16]: It is just to keep the implementation details hidden from the caller. In future if I want to change the authentication mechanism, I can easily do that without affecting the other parts of application. Don't you think, this is fine ?

File: app/build.gradle.kts (hunk 1 above)
Comment (doersweb) [reply app/build.gradle.kts]: Updated, thank you.

File: app/src/main/java/com/doersweb/tasky/interactor/authentication/AuthenticationInteractorImpl.kt (hunk 2 above)
Comment (philipplackner) [reply app/src/main/java/com/doersweb/tasky/interactor/authentication/AuthenticationInteractorImpl.kt:16]: But do these contain data logic or business logic?

**Pull request merged**